colossalai/inference/tensor_parallel/engine.py

```diff
@@ -243,14 +357,14 @@ def _generate_by_set_infer_state(self, input_tokens, **generate_kwargs) -> torch
         # NOTE this is not a preferable way to pass BatchInferState during inference
         #   we might want to rewrite generate function (e.g. _generate_by_pass_infer_state)
         #   and pass BatchInferState via model forward
-        model = self.model
+        model = self.sharded_model
         if isinstance(model, LlamaForCausalLM):
-            model = self.model.model
+            model = self.sharded_model.model
         elif isinstance(model, BloomForCausalLM):
-            model = self.model.transformer
+            model = self.sharded_model.transformer
         setattr(model, 'infer_state', batch_infer_state)
 
-        outputs = self.model.generate(**input_tokens, **generate_kwargs, early_stopping=False)
+        outputs = self.sharded_model.generate(**input_tokens, **generate_kwargs, early_stopping=False)
```

Comment:
For condition of prompt_token_ids  is not None in `generate`, input_tokens will be set as prompt_token_ids: `input_tokens = prompt_token_ids`, and then `**input_tokens` here will error out, since it's not a dictionary.

Reply:
Done